satellite/audit/worker.go

```diff
@@ -78,6 +102,45 @@ func (worker *Worker) process(ctx context.Context) (err error) {
 }
 
 func (worker *Worker) work(ctx context.Context, path storj.Path) error {
-	// TODO: handle work.
-	return nil
+	var errlist errs.Group
+	report, err := worker.verifier.Reverify(ctx, path)
+	if err != nil {
+		errlist.Add(err)
+	}
+
+	// TODO(moby) we need to decide if we want to do something with nodes that the reporter failed to update
+	_, err = worker.reporter.RecordAudits(ctx, report)
+	if err != nil {
+		errlist.Add(err)
+	}
+
+	// Skip all reverified nodes in the next Verify step.
+	skip := make(map[storj.NodeID]bool)
+	if report != nil {
+		for _, nodeID := range report.Successes {
+			skip[nodeID] = true
+		}
+		for _, nodeID := range report.Offlines {
+			skip[nodeID] = true
+		}
+		for _, nodeID := range report.Fails {
+			skip[nodeID] = true
+		}
+		for _, pending := range report.PendingAudits {
+			skip[pending.NodeID] = true
+		}
+	}
+
+	report, err = worker.verifier.Verify(ctx, path, skip)
```

Comment:
nit - add another comment `Next, audit the the remaining nodes that are not in containment mode`